File: leetcode/python/1436_destination_city.py
class Solution:
    #  [["B","C"],["D","B"],["C","A"]]
    def destCity(self, paths: list[list[str]]) -> str:
        dest = set[str]()
        points = set[str]()
        
        for u, v in paths:
            points.add(u)
            if u in dest:
                dest.remove(u)
            if v not in points:
                dest.add(v)
        return dest.pop()
            
        

File: leetcode/rust/src/1436_destination_city.rs
use std::collections::HashSet;

impl Solution {

    pub fn dest_city(paths: Vec<Vec<String>>) -> String {
        let mut set = HashSet::new();
        for path in paths.iter() {
            set.insert(&path[0]);
        }
        for path in paths.iter() {
            if !set.contains(&path[1]) {
                return path[1].clone();
            }
        }
        String::new()
    }
}